### PR TITLE
feat(bulk2): allow to get raw job results

### PR DESCRIPTION
**Pull request by cristiand391**
Allow to fetch raw CSV results from bulk2 jobs.

The getResults method would fetch the CSV, parse them into JS arrays and cache that in an internal class field, this PR adds support for these methods to return results a CSV (string) and intentionally skip caching to avoid switching types.

Required for:
https://github.com/salesforcecli/plugin-data/pull/1097

@W-12408034@

### CODE REVIEW

File: src/api/bulk2.ts [hunk 1]
```diff
@@ -863,7 +863,21 @@ export class IngestJobV2<S extends Schema> extends EventEmitter {
...
+  async getSuccessfulResults(raw?: boolean): Promise<IngestJobV2SuccessfulResults<S> | string> {
+    const reqOpts: BulkRequest = {
+      method: 'GET',
+      path: `/${this.id}/successfulResults`,
+    }
+
+    if (raw) {
+      return this.createIngestRequest<string>({
+        ...reqOpts,
+        responseType: 'text/plain',
```
Comment (cristiand391) [inline src/api/bulk2.ts:877]: jsforce will check `responseType` ( or fallback to `content-type` header ) to try parse the body request to a JS value, passing `text/plain` so it skip parsing.

File: src/api/bulk2.ts [hunk 2]
```diff
@@ -887,16 +901,29 @@ export class IngestJobV2<S extends Schema> extends EventEmitter {
    *
    * @returns Promise<IngestJobV2SuccessfulResults>
```
Comment (WillieRuemmele) [inline src/api/bulk2.ts:902]: should we update jsdocs to mention this parameter?

**Review by WillieRuemmele — APPROVED** [review]

**Comment by WillieRuemmele** [comment]
[QA](https://github.com/salesforcecli/plugin-data/pull/1097#issuecomment-2427710275) ✅

**Review by iowillhoit — APPROVED** [review]
LG

**Pull request merged**